src/codegen/intermediate_representation.c

```diff
@@ -652,11 +645,23 @@ IRInstruction *ir_create_static
   return ref;
 }
 
-IRInstruction *ir_stack_allocate
-(CodegenContext *context,
- Type *type
- )
-{
+IRInstruction *ir_static_reference(CodegenContext *context, span name) {
```

Comment:
What is "the variable itself" here?

Reply:
The `IRStaticVariable*` or whatever it’s called

Reply:
How would we go about getting the IRStaticVariable from a variable reference AST node?

Reply:
See one of my other comments in codegen.c just now

Reply:
As it turns out, now that the backend is working better with order of operands and such, we can also just use `subs_lhs = var` instead of `subs_lhs = ir_static_reference(...)`. So we may actually be able to remove this.

Tangent: doing this has made me realise what stack spilling is and why it's needed; you can save a value on the stack and then just load it into a register vs keeping that register around for as long as it's needed, freeing up more registers for other things.

Reply:
https://github.com/LensPlaysGames/FUNCompiler/blob/6411d364fb55d1f81fff0a5f33ee85b3188a2976/src/codegen.c#L376-L380

Reply:
> As it turns out, now that the backend is working better with order of operands and such, we can also just use `subs_lhs = var` instead of `subs_lhs = ir_static_reference(...)`

Yeah, in that branch, the `var` is already a static ref, so you can just reuse that one. That’s pretty much what I meant with ‘you don’t have to look up variables by name’.

> ```c
> if (var->kind == IR_STATIC_REF) 
>      subs_lhs = var; 
> else if (var->kind == IR_ALLOCA) 
>      subs_lhs = var;
> ```

I think this could just be
```c
if (var->kind == IR_STATIC_REF || var->kind == IR_ALLOCA)
    subs_lhs = var;
```
;Þ